pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java

```diff
@@ -64,12 +66,19 @@
     ARRAY_MAP_BASED
   }
 
+  // Enum to reflect if trimming of group keys is ON or OFF
+  private enum TrimMode {
```

Comment:
No, this is being used to keep track of what ids have been removed. When we enter TrimMode.ON, we will start tracking removed groupKeys. When purge is called, we need this mapping from id to group keys to find what group keys to purge.